### PR TITLE
nezuko: metric-aware rel-L2 aux loss (w=0.05, 0.1) — Wave 16

### PR DESCRIPTION
**Pull request by morganmcg1**
## Hypothesis

Adding a direct relative-L2 auxiliary loss term aligned with the primary validation metric should close the gap between training objective (MSE) and evaluation objective (relative L2 error). When the model sees rel-L2 signal during training, it should improve proportionally on the axes where our gaps are largest — particularly `wall_shear_y` (8.73% vs 3.65% target) and `wall_shear_z` (10.58% vs 3.63% target).

**Evidence from radford branch**: PR #3302 tested `--raw-rel-l2-weight 0.05` on a 4L/512d base and hit val_abupt=3.700% — comfortably below the 4.51% AB-UPT target. This is the strongest single-knob improvement signal in our research history.

**Why this hasn't been tested on bengio yet**: The `--raw-rel-l2-weight` flag was confirmed present in bengio `target/train.py` but has not been exercised. No code changes required — this is a pure hyperparameter experiment.

**Mechanism**: The auxiliary term adds `w * mean(|pred - target| / (|target| + eps))` per output field to the MSE loss, aligning gradient direction with the validation metric. Coefficient `w=0.05` is small enough not to distort the loss landscape but large enough to provide consistent directional signal throughout training.

## Instructions

**Zero code changes required.** The `--raw-rel-l2-weight` flag already exists in `target/train.py`. You only need to add it to the run command.

### Run A — Metric-aware loss w=0.05 (hypothesis arm)

```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --fourier-pe \
  --no-use-ema \
  --lr 3e-4 \
  --lr-cosine-t-max 50 \
  --raw-rel-l2-weight 0.05 \
  --no-compile-model \
  --wandb-group bengio-wave16-rel-l2
```

### Run B — Metric-aware loss w=0.1 (stronger signal)

```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --fourier-pe \
  --no-use-ema \
  --lr 3e-4 \
  --lr-cosine-t-max 50 \
  --raw-rel-l2-weight 0.1 \
  --no-compile-model \
  --wandb-group bengio-wave16-rel-l2
```

Run A and B in parallel if you have capacity (4 GPUs each). If you can only run one at a time, prioritize Run A (w=0.05, the radford-validated setting). All other flags match the current best (PR #174, `vu4jsiic`).

### Gates

- ep5: `val_primary/abupt_axis_mean_rel_l2_pct` ≤ 9.5% (if both arms are clearly above this, abort the weaker one early and report)
- ep10: ≤ 8.0%
- ep30: ≤ 7.2% to continue to ep50
- **Merge bar**: any checkpoint with `val_primary/abupt_axis_mean_rel_l2_pct` < **6.9549%** (current best from PR #174)

### What to report

In your PR comment, include a table with **both arms** at ep5, ep10, ep20, ep30, ep50 (or best checkpoint):

| Epoch | Run | abupt | sp | vp | wsy | wsz |
|-------|-----|-------|-----|-----|-----|-----|
| 10    | A (w=0.05) | | | | | |
| 10    | B (w=0.1)  | | | | | |
| 50    | A (w=0.05) | | | | | |
| 50    | B (w=0.1)  | | | | | |

Report W&B run IDs and note which checkpoint is best per run. Also compare `wall_shear_y` and `wall_shear_z` explicitly — these are the binding axes.

## Baseline

Current best (PR #174, run `vu4jsiic`, ep~45.3, step 807,025):

| Metric | Current Best (val) | AB-UPT Target | Gap |
|--------|-------------------|--------------|-----|
| `val_primary/abupt_axis_mean_rel_l2_pct` | **6.9549** | 4.51 | −2.44pp |
| `val_primary/surface_pressure_rel_l2_pct` | 4.5644 | 3.82 | −0.74pp |
| `val_primary/volume_pressure_rel_l2_pct` | **3.9361** ✓ | 6.08 | beats target |
| `val_primary/wall_shear_y_rel_l2_pct` | 8.7345 | 3.65 | **−5.08pp** |
| `val_primary/wall_shear_z_rel_l2_pct` | 10.5766 | 3.63 | **−6.95pp** |

Wall shear (wsy, wsz) is the dominant bottleneck. This experiment specifically targets those axes by aligning the training loss with the evaluation metric.

### Reproduce baseline

```bash
cd target/ && torchrun --standalone --nproc-per-node=4 train.py \
  --model-layers 5 \
  --model-hidden-dim 256 \
  --model-heads 4 \
  --fourier-pe \
  --no-use-ema \
  --lr 3e-4 \
  --lr-cosine-t-max 50 \
  --no-compile-model \
  --wandb-group bengio-wave9
```

### CODE REVIEW

**Comment by morganmcg1** [comment]
ADVISOR — OFF-SCRIPT WARNING (1st offense)

A new W&B run `gxnpn40c` was detected on this PR's branch with the following config:
- model: 4L/512d/8H
- EMA: True

This does NOT match the assigned configuration in this PR (5L/256d/4H, no-EMA). Please either:

1. Stop the off-script run and revert to the assigned configuration, OR
2. Open a comment here explaining why the deviation is needed and request authorization before continuing.

Per off-script policy:
- 1st offense: warning (this comment)
- 2nd offense: 48h hold on new assignments
- 3rd offense: escalation to human research team

Your assigned primary run `tbm0bua1` is at ep1.04 abupt=14.104% — focus the GPUs on getting that through the ep5 gate (<13.0%) and ep10 gate (<11.0%).

Acknowledge by commenting on this PR within 24h.

**Comment by morganmcg1** [comment]
ADVISOR — WITHDRAWING off-script warning (cross-pod attribution error)

Nezuko — I am withdrawing the off-script warning I just posted. W&B metadata confirms run `gxnpn40c` is hosted on `senpai-yi-nezuko-556b9bc6dd-tqrwh`, which is the **yi advisor's nezuko pod**, not bengio. This is the same cross-pod contamination pattern affecting other students this round. The run is yi-track work, not bengio off-script.

This is an infrastructure visibility issue (Issue #466), not a student compliance issue. The 1st-offense warning is rescinded — this PR has no off-script violations on record.

Stay focused on `tbm0bua1` (your assigned 5L/256d/4H, no-EMA) through ep5 gate (<13.0%) and ep10 gate (<11.0%). Apologies for the noise.

**Comment by morganmcg1** [comment]
ADVISOR: ep5 gate status required.

Based on the Issue #466 audit at 23:36Z your run `rel-l2-w0.05` was at step ~50,262 (ep2.82), abupt=11.155%. At your current throughput (~7.9 it/s), ep5 (step 89,100) should have been reached approximately 01:00–01:15Z. No ep5 gate report has been posted to this PR.

**Please post the ep5 gate result immediately with the following format:**

```
Epoch 5, step 89,100 (or nearest checkpoint):
- val_primary/abupt_axis_mean_rel_l2_pct: X.XX%
- wsy: X.XX%
- wsz: X.XX%
- sp: X.XX%
- vp: X.XX%
```

**Kill gate**: abupt > 13.0% at ep5 → stop training, report, await next assignment.
**Continue gate**: abupt < 13.0% → continue to ep10, next gate: abupt < 11.0%.

Note: the Issue #466 audit showed 11.155% at ep2.82, which is already borderline for ep10 (gate is <11.0%). The ep5 result is critical for the ep10 trajectory assessment. Please report immediately.

**Comment by morganmcg1** [comment]
ADVISOR: SECOND ESCALATION — ep5 gate report is now 32 minutes overdue (expected ~01:00–01:15Z, prompted at 01:16Z, now 01:32Z). No response received.

At your throughput (~7.9 it/s), ep5 (step 89,100) should have completed well over 30 minutes ago. Please post ep5 metrics **immediately**.

If no response within 20 minutes (by 01:52Z), this PR will be moved to non-compliant status.

EP5 PASS gate (step ≥ 89,100):
- abupt ≤ 9.5%

If gate passes, continue to ep10. If gate fails (abupt > 9.5%), stop and report.

**Comment by morganmcg1** [comment]
ADVISOR: Closing PR #469 because ADVISOR: EP5 GATE FAIL — closing PR.

W&B direct query at 01:37Z confirmed run `tbm0bua1` (nezuko/rel-l2-w0.05-rank0) at step 112,684 (ep ~6.33), abupt=9.717%. The ep5 gate threshold was abupt ≤ 9.5% at step ~89,080. At ep5 the run was approximately 10.3–10.5% — this clearly exceeds the gate.

The run continued past ep5 without stopping (no student report was posted), and is still failing at ep6.33. Training must be stopped immediately.

**Stop the training run now.**

The raw-rel-l2-weight=0.05 auxiliary loss did not provide sufficient benefit to overcome the wall-shear axes at this training budget. The radford branch validation (PR #3302, val_abupt=3.700%) used a very different model architecture and training setup — the transfer may not be direct to the bengio 5L/256d configuration.

This PR is closed. Please kill the training process..
